mmv1/templates/terraform/constants/apphub_application.go.tmpl

```diff
@@ -0,0 +1,17 @@
+func apphubApplicationCustomizeDiff(_ context.Context, diff *schema.ResourceDiff, meta interface{}) error {
+    if diff.HasChange("location") || diff.HasChange("scope.0.type") {
+        location := diff.Get("location")
+        scope_type := diff.Get("scope.0.type")
+
+        if scope_type == "GLOBAL" {
+            if location != "global" {
+                return fmt.Errorf("Error validating location %s with %s scope type", location, scope_type)
+            }
+        } else {
+            if location == "global" {
+                return fmt.Errorf("Error validating location %s with %s scope type", location, scope_type)
+            }
+        }
+    }
+    return nil
+}
```

Comment:
we should cover this logic somewhere in a test. If we are going to keep it (see my previous comment), let's add an invalid config in https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/apphub/resource_apphub_application_test.go and use [ExpectError](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/servicenetworking/resource_service_networking_connection_test.go#L83)

Reply:
Done